src/com/google/javascript/jscomp/TranspilationPasses.java

```diff
@@ -112,6 +112,7 @@ static void addPreTypecheckTranspilationPasses(
               Feature.REGEXP_FLAG_U,
               Feature.REGEXP_FLAG_Y));
       passes.add(es6NormalizeShorthandProperties);
+      passes.add(es6RewriteClassExtends);
```

Comment:
Why here instead of immediate before class rewriting? Alternately, why isn't it part of class rewriting?

Reply:
See the comment at https://github.com/google/closure-compiler/pull/2997/files#diff-fa51555d8bc30554f2cbba232a8033bdR41

This must be performed before transpilation of `super` calls. So much of class rewriting depends on the extends clause being a qualified name.